python/tests/test_slicebyvalue.py

```diff
@@ -0,0 +1,103 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (c) 2020 Scipp contributors (https://github.com/scipp)
+# @file
+# @author Matthew Andrew
+import scipp as sc
+import numpy as np
+import pytest
+
+
+class TestSliceByValue:
+    def setup_method(self):
+        var = sc.Variable(['x'], values=np.arange(5, dtype=np.float) + 0.5)
+        values_a = sc.Variable(dims=['x'],
+                               values=[1.0, 1.1, 1.2, 1.3, 1.4],
+                               unit=sc.units.m)
+        values_b = sc.Variable(dims=['x'],
+                               values=[1.0, 2.0, 3.0, 4.0, 5.0],
+                               unit=sc.units.m)
+        self._d = sc.Dataset(data={
+            'a': values_a,
+            'b': values_b
+        },
+                             coords={'x': var})
+
+    def test_slice_by_single_value(self):
+        by_value = self._d['a']['x', 1.5 * sc.units.dimensionless]
+        by_index = self._d['a']['x', 1]
+        assert sc.is_equal(by_value, by_index)
+
+    def test_assigning_to_slice_by_value(self):
+        self._d['a']['x', 1.5 * sc.units.dimensionless] = 5.7 * sc.units.m
+        slice = self._d['a']['x', 1.5 * sc.units.dimensionless].values
+        assert slice == np.array(5.7)
+
+    def test_modifying_slice_in_place(self):
+        self._d['a']['x', 1.5 * sc.units.dimensionless] *= 2.5
+        slice = self._d['a']['x', 1.5 * sc.units.dimensionless].values
+        assert slice == np.array(2.75)
+
+        self._d['a']['x', 1.5 * sc.units.dimensionless] += 2.25 * sc.units.m
+        slice = self._d['a']['x', 1.5 * sc.units.dimensionless].values
+        assert slice == np.array(5.0)
+
+        self._d['a']['x', 1.5 * sc.units.dimensionless] -= 3.0 * sc.units.m
+        slice = self._d['a']['x', 1.5 * sc.units.dimensionless].values
+        assert slice == np.array(2.0)
+
+        self._d['a']['x', 1.5 * sc.units.dimensionless] /= 2.0
+        slice = self._d['a']['x', 1.5 * sc.units.dimensionless].values
+        assert slice == np.array(1.0)
+
+    def test_slice_with_range(self):
+        by_value = self._d['a']['x', 1.5 * sc.units.dimensionless:4.5 *
```

Comment:
```suggestion
        by_value = self._d['a']['x', 1.5 * sc.units.dimensionless:4.5,
```
and similar below?

Reply:
With the current implementation I need both ends of the range to have units. Do we want to change it so only one end of the range has units and we assume the same units for the other?

Reply:
No, both should have units.

Reply:
But my comment was not about that: You used `*` so you are no range-slicing.

Reply:
I think this is range slicing here. The implementation has switched to using the py::slice format so the syantax
by_value = self._d['a']['x', 1.5 * sc.units.dimensionless:4.5 * sc.units.dimensionless] should be doing a slice between 1.5 and 4.5.  The assert statement below checks this is equal to the same slice done by index.

Reply:
Yes, but you have a typo: `*` instead of `:`.

Reply:
At least that is what by github diff shows?

Reply:
It looks correct when looking at the file, must be a glitch. -> Ignore this.